go-controller/pkg/ovn/master.go

```diff
@@ -908,5 +994,73 @@ func (oc *Controller) syncNodes(nodes []interface{}) {
 		if err := oc.deleteNode(nodeName, nodeSubnets.hostSubnet, nodeSubnets.joinSubnet); err != nil {
 			klog.Error(err)
 		}
+		//remove the node from the chassis map so we don't delete it twice
+		delete(chassisMap, nodeName)
+	}
+
+	for nodeName, chassisName := range chassisMap {
+		if chassisName != "" {
+			_, stderr, err = util.RunOVNSbctl("--if-exist", "chassis-del", chassisName)
+			if err != nil {
+				klog.Errorf("Failed to delete chassis with name %s for logical switch %s: stderr: %q, error: %v",
+					chassisName, nodeName, stderr, err)
+			}
+		}
+	}
+}
+
+type chassisData struct {
+	nodeName    string
+	chassisName string
+}
+
+func (oc *Controller) unmarshalChassisDataIntoMap(chData []byte) (map[string]string, error) {
+	//map of node name to chassis name
+	chassisMap := make(map[string]string)
+	var mapUnmarshal map[string][]interface{}
+
+	if len(chData) == 0 {
+		return chassisMap, nil
+	}
+
+	err := json.Unmarshal(chData, &mapUnmarshal)
+	if err != nil {
+		klog.Errorf("Error while unmarshaling received chassis data")
+		return chassisMap, err
+	}
+
+	if data, ok := mapUnmarshal["data"]; !ok {
```

Comment:
I think rather than doing all this unmarshal, marshal, and unmarshal again you can just simplify all of this unmarshal one time. For example:

`	chassisMap := make(map[string]string)
        chData := []byte(`{"data":[["1553f2c3-6c71-4bf1-b67d-d318d19e5101","ovn-control-plane"]],"headings":["name","hostname"]}`)
	// map of data to slices of str slices
	var mapUnmarshal map[string][][]string

	json.Unmarshal(chData, &mapUnmarshal)
	if data, ok := mapUnmarshal["data"]; ok {
		for _, entry := range data {
			if len(entry) < 2 || entry[0] == "" || entry[1] == "" {
			  continue
			}
			chassisMap[entry[0]] = entry[1]
		}
	}
        fmt.Println(chassisMap)`

Outputs:
map[1553f2c3-6c71-4bf1-b67d-d318d19e5101:ovn-control-plane]

Reply:
I guess I could do that, but it is very fragile IMO, in that it breaks if for whatever reason each entry is not a slice. I am more in the favor of completeness here over assumption.

Reply:
Although, I guess we are hardcoding some of the things like looking for `"data"` in the json returned, so in that sense, it is pretty specific. Hmm. I am OK either ways.